kivy/uix/behaviors/focus.py

```diff
@@ -460,10 +458,49 @@ def _handle_post_on_touch_up(touch):
                 continue
             focusable.focus = False
 
-    def _get_focus_next(self, focus_dir):
+    def get_focus_next(self):
+        '''This method provides the next focuses(ie self.focus_next)
+        and ponders over two cases if one hits focusable we walk through
+        focus_next.If one hits unfocusable we walk the widget tree
+        keeping various cases in mind in both of the cases.
+        '''
         current = self
-        walk_tree = 'walk' if focus_dir is 'focus_next' else 'walk_reverse'
+        walk_tree = 'walk'
+        focus_dir = 'focus_next'
+        while 1:
```

Comment:
Code should not be duplicated because it leads to maintenance issues. The way it was before, except with an added return and documentation is how it should be. That required no code duplication.